From 685ec336f58c0dd6e78bf721c90c3b907b7e500c Mon Sep 17 00:00:00 2001
From: Patrick Delaunay <patrick.delaunay@foss.st.com>
Date: Fri, 19 Jul 2024 11:34:45 +0200
Subject: [PATCH 665/954] refactor(stm32mp2): simplify STM32MP2 flag selection
 in platform.mk

Simplify the STM32MP2X flag in makefile and detect device tree
used for other SoC.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 55f1bf84e0ef32362859e2c424776049cd5044aa refactor(stm32mp2): simplify STM32MP2 flag selection in platform.mk]
Signed-off-by: Patrick Delaunay <patrick.delaunay@foss.st.com>
Change-Id: I6986c378a0b3be617b92c7f9a65c12d7aef1a60f
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/394274
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
Tested-by: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 plat/st/stm32mp2/platform.mk | 40 ++++++++++--------------------------
 1 file changed, 11 insertions(+), 29 deletions(-)

diff --git a/plat/st/stm32mp2/platform.mk b/plat/st/stm32mp2/platform.mk
index 3739865b66..961fae338d 100644
--- a/plat/st/stm32mp2/platform.mk
+++ b/plat/st/stm32mp2/platform.mk
@@ -39,40 +39,22 @@ STM32MP23			?=	0
 STM32MP25			?=	0
 STM32MP_M33_TDCID		?=	0
 
-ifeq ($(STM32MP21),1)
-ifeq ($(STM32MP23),1)
-$(error Cannot enable both flags STM32MP21 and STM32MP23)
-endif
-ifeq ($(STM32MP25),1)
-$(error Cannot enable both flags STM32MP21 and STM32MP25)
-endif
+ifneq ($(findstring stm32mp21,$(DTB_FILE_NAME)),)
 STM32MP21			:=	1
-STM32MP23			:=	0
-STM32MP25			:=	0
-else ifeq ($(STM32MP23),1)
-ifeq ($(STM32MP25),1)
-$(error Cannot enable both flags STM32MP23 and STM32MP25)
 endif
-STM32MP21			:=	0
-STM32MP23			:=	1
-STM32MP25			:=	0
-else ifeq ($(STM32MP25),1)
-STM32MP21			:=	0
-STM32MP23			:=	0
-STM32MP25			:=	1
-else ifneq ($(findstring stm32mp21,$(DTB_FILE_NAME)),)
-STM32MP21			:=	1
-STM32MP23			:=	0
-STM32MP25			:=	0
-else ifneq ($(findstring stm32mp23,$(DTB_FILE_NAME)),)
-STM32MP21			:=	0
+ifneq ($(findstring stm32mp23,$(DTB_FILE_NAME)),)
 STM32MP23			:=	1
-STM32MP25			:=	0
-else ifneq ($(findstring stm32mp25,$(DTB_FILE_NAME)),)
-STM32MP21			:=	0
-STM32MP23			:=	0
+endif
+ifneq ($(findstring stm32mp25,$(DTB_FILE_NAME)),)
 STM32MP25			:=	1
 endif
+ifneq ($(filter 1,$(STM32MP21) $(STM32MP23) $(STM32MP25)), 1)
+$(warning STM32MP21=$(STM32MP21))
+$(warning STM32MP23=$(STM32MP23))
+$(warning STM32MP25=$(STM32MP25))
+$(warning DTB_FILE_NAME=$(DTB_FILE_NAME))
+$(error Cannot enable 2 flags STM32MP2X)
+endif
 
 STM32MP_USE_EXTERNAL_HEAP 	:=	1
 
-- 
2.53.0

